File: Arcade/TheCore/Java/IsSmooth.java
final class IsSmooth {
	
	boolean isSmooth(int[] arr) {
	    int l = arr.length;
	    
	    if (l == 1) return true;
	    
	    int middle = switch (l % 2) {
	        case 0 -> arr[l / 2] + arr[l / 2 - 1];
	        default -> arr[l / 2];
	    };
	    
	    return arr[0] == middle && arr[l - 1] == middle;
	}
}

File: Arcade/TheCore/Kotlin/isSmooth.kt
fun isSmooth(arr: MutableList<Int>): Boolean {
    val l: Int = arr.size
    
    if (l == 1) return true
    
    var middle: Int = when (l % 2) {
        0 -> arr[l / 2] + arr[l / 2 - 1]
        else -> arr[l / 2]
    }
    
    return arr[0] == middle && arr.last() == middle
}